tokio/src/runtime/task/core.rs

```diff
@@ -160,7 +160,7 @@ impl<T: Future, S: Schedule> Core<T, S> {
                 let waker_ref = waker_ref::<T, S>(header);
                 let mut cx = Context::from_waker(&*waker_ref);
 
-                crate::coop::budget(|| future.poll(&mut cx))
+                future.poll(&mut cx)
```

Comment:
I have some vague recollection of there being an additional place that ended up calling poll through here beyond `basic_scheduler`, `threaded_scheduler`, and `local_set`. Maybe related to `block_on`? Would it make sense to (temporarily?) add a `debug_assert` here that budgeting is enabled just to make sure?

Reply:
I can... block_on doesn't run through a task though. IIRC local set, basic, threaded, and blocking are the only locations.

I did not add it to blocking as blocking is just using the Task structure for the join handle.

Reply:
i tried adding the assertion but it broke a miri test. miri doesn't have thread locals. I opted to revert the assertion instead of doing a more significant reorg.

Reply:
I'm satisfied if the non-loom tests ran without problem with the assertion there.